lightning/src/onion_message/messenger.rs

```diff
@@ -298,11 +408,15 @@ fn outbound_buffer_full(peer_node_id: &PublicKey, buffer: &HashMap<PublicKey, Ve
 	false
 }
 
-impl<ES: Deref, NS: Deref, L: Deref, CMH: Deref> OnionMessageHandler for OnionMessenger<ES, NS, L, CMH>
-	where ES::Target: EntropySource,
-		  NS::Target: NodeSigner,
-		  L::Target: Logger,
-		  CMH::Target: CustomOnionMessageHandler + Sized,
+impl<ES: Deref, NS: Deref, L: Deref, MR: Deref, OMH: Deref, CMH: Deref> OnionMessageHandler
+for OnionMessenger<ES, NS, L, MR, OMH, CMH>
+where
+	ES::Target: EntropySource,
+	NS::Target: NodeSigner,
+	L::Target: Logger + Sized,
```

Comment:
`Sized` here is causing problems with a `lightning-background-processor` doc test with `--features=futures` that uses `Arc<dyn lightning::util::logger::Logger + Send + Sync>`.

```
error[E0277]: the size for values of type `(dyn Logger + Send + std::marker::Sync + 'static)` cannot be known at compilation time
   --> src/lib.rs:529:295
    |
34  | ...MyChannelManager>, my_gossip_sync: Arc<MyGossipSync>, my_logger: Arc<MyLogger>, my_scorer: Arc<MyScorer>, my_peer_manager: Arc<MyPeerManager>) {
    |                                                                                                                               ^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `Sized` is not implemented for `(dyn Logger + Send + std::marker::Sync + 'static)`
    = note: required because of the requirements on the impl of `OnionMessageHandler` for `OnionMessenger<Arc<KeysManager>, Arc<KeysManager>, Arc<(dyn Logger + Send + std::marker::Sync + 'static)>, IgnoringMessageHandler, IgnoringMessageHandler>`
note: required by a bound in `PeerManager`
   --> /Users/jkczyz/src/rust-lightning/lightning/src/ln/peer_handler.rs:683:15
    |
683 |         OM::Target: OnionMessageHandler,
    |                     ^^^^^^^^^^^^^^^^^^^ required by this bound in `PeerManager`

```

The problem seems to be that `handle_onion_message` calls methods that take `&L` and `where L: Logger`.

@TheBlueMatt I can fix this by having those methods take `&L` and `where L: Deref, L::Target: Logger`. Just seems weird to have a reference to a `Deref`. Alternatively,  keeping `L`  and using `where L: Deref + Clone, L::Target: Logger` works since both `&` and `Arc` implement `Clone`. Not sure what the canonical way of doing this is, though.

Reply:
Thanks, that did the trick. Removed the other `Sized` bounds on the the block while I was at it.